chore: pydantic 2.11

sfoster1 opened the pull request:
Bumps the system to use pydantic 2.11, which has much less memory usage when building model classes.

Pretty well tested on flex and ot-2 (ot-2 requires the buildroot pr).

sfoster1 committed:
chore: pydantic 2.11

sfoster1 committed:
small fixup

SyntaxColoring commented on api/src/opentrons/protocol_engine/types/module.py:
```diff
@@ -197,7 +197,7 @@ class ModuleDefinition(BaseModel):
     )
     gripperOffsets: Optional[Dict[str, LabwareMovementOffsetData]] = Field(
-        default_factory=dict,
+        default_factory=lambda: {},
```
Hm, what was wrong with `dict`? We definitely have a bunch of other places that are doing `default_factory=list`.

sfoster1 replied:
idk but mypy yelled at me

SyntaxColoring commented on robot-server/Makefile:
```diff
@@ -64,7 +64,7 @@ clean_all_cmd = $(clean_cmd) dist
 dev_port ?= "31950"
 dev_host ?= "localhost"
-run_dev ?= uvicorn "robot_server.app:app" --host $(dev_host) --port $(dev_port) --ws wsproto --lifespan on --reload --reload-dir robot_server
+run_dev ?= memray run -q --live-remote --live-port 12345 -m uvicorn "robot_server.app:app" --host $(dev_host) --port $(dev_port) --ws wsproto --lifespan on --reload --reload-dir robot_server
```
Is this meant to be in here?

sfoster1 replied:
definitely not

SyntaxColoring commented on api/Pipfile.lock:
This is introducing a new transitive dependency of Pydantic, `typing-inspection`. Does that need to be pinned in `Pipfile` to keep it from outpacing the version that's hard-coded in Opentrons/buildroot?

sfoster1 replied:
I don't think so; it's part of the pydantic project, and we don't explicitly pin `pydantic-core` in our pipfiles.

SyntaxColoring reviewed (changes requested):
Looks good ~~once tests pass.~~ Changes requested for a Makefile change that I think is accidental.

sfoster1 committed:
didnt mean to commit this

SyntaxColoring commented on robot-server/Makefile:
```diff
@@ -64,7 +64,7 @@ clean_all_cmd = $(clean_cmd) dist
 dev_port ?= "31950"
 dev_host ?= "localhost"
-run_dev ?= uvicorn "robot_server.app:app" --host $(dev_host) --port $(dev_port) --ws wsproto --lifespan on --reload --reload-dir robot_server
+run_dev ?= $(python) -m uvicorn "robot_server.app:app" --host $(dev_host) --port $(dev_port) --ws wsproto --lifespan on --reload --reload-dir robot_server
```
I think we just want `uvicorn`, not `$(python) -m uvicorn`.

This is used like `$(pipenv) run $(run_dev)`, which would expand to something like `python -m pipenv run pipenv run python -m uvicorn`, with the extra nested `pipenv run`.

sfoster1 replied:
good catch, i think i got confused because we run with -m in the systemd unit on the robot

sfoster1 committed:
whoops

SyntaxColoring reviewed (approved):
Thanks!

sfoster1 committed:
some schem updates

sfoster1 committed:
format

sfoster1 committed:
upsetting

SyntaxColoring committed:
Merge branch 'edge' into pydantic-2-11

Resolve conflicts in Pipenv.lock files in api, hardware, robot-server, shared-data, and system-server.

Rerun `pipenv lock` in those directories. Fails for system-server.

SyntaxColoring committed:
Fix system-server's pytest-asyncio spec and re-lock.

A recent dependency upgrade changed the spec to pytest-asyncio>=1, but it should have been >=0.23.0 because >=1 is incompatible with pytest<8. That was already fixed for other projects, but somehow it snuck through for this one.

After fixing that, `pipenv lock --dev`.

SyntaxColoring commented on system-server/Pipfile:
```diff
@@ -29,7 +29,7 @@ flake8-annotations = "~=3.0.1"
 flake8-noqa = "~=1.4.0"
 pytest = "<8,>=7.2.2"
-pytest-asyncio = "<2,>=1"
+pytest-asyncio = "<2,>=0.23.0"
```
This change is intentional. `>=1` was unresolvable. It snuck into `edge` somehow through an earlier PR of mine.
